docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+
+It is ***NOT*** in the scope of the **RPC Companion**:
+
+1. Provide an authentication mechanism for the RPC Companion endpoint.
+
+### [RPC Endpoint](#rpc-endpoint)
+
+The RPC Companion endpoint will be the same as the CometBFT endpoint but with a `/v1` appended to it. The RPC Companion endpoint
+might also use a different port than the default CometBFT RPC port (e.g. `26657`).
+
+For example, suppose these are the URLs for each RPC endpoint:
+
+CometRPC -> `http://cosmos.host:26657`
+
+RPC Companion -> `http://rpc-companion.host:8080/v1`
+
+To make a request for a `block` at height `5` using the CometBFT endpoint:
+
+`curl --header "Content-Type: application/json" --request POST --data '{"method": "block", "params": ["5"], "id": 1}' http://cosmos.host:26657`
+
+The make the same request to the RPC Companion endpoint:
+
+`curl --header "Content-Type: application/json" --request POST --data '{"method": "block", "params": ["5"], "id": 1}' http://rpc-companion.host:8080/v1`
+
+> Note that only the URL changes between these two `curl` commands
+
+The RPC Companion will accept JSONRPC requests, the same way as the CometBFT RPC endpoint does.
+
+The methods for the RPC Companion endpoint on the following table are the ones that can be implemented first since it seems it will be straightforward and less complex
+
+| **JSONRPC method**   | **JSONRPC Parameters**                 | **Description**                                 | **Notes**                                                                                                                                                                                                                                                                               |
+|----------------------|----------------------------------------|-------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `abci_info`          |                                        | Get information about the application           | This companion endpoint will return the same response structure as the CometBFT endpoint. The companion endpoint will return the latest information stored in its database that was retrieved from the full node.                                                                       |
+| `block`              | * height                               | Get block at a specified height                 | This companion endpoint will return the same response structure as the CometBFT endpoint. The data retrieved from the companion database for a particular block will have to be properly serialized into the `block` struct in order to be returned as a response.                      |
```

Comment:
It appears that all methods in this table will return the same response as the CometBFT RPC corresponding endpoint. Maybe it is good do add this information to the table caption, instead of repeating it on each line.